todo.sh

```diff
@@ -26,6 +26,25 @@ export TODO_SH TODO_FULL_SH
 
 oneline_usage="$TODO_SH [-fhpantvV] [-d todo_config] action [task_number] [task_description]"
 
+# Assumption: The in-place argument is the first
+# Assumption: Only a single file is processed with sed
+sed() {
```

Comment:
What if instead of `TODOTXT_SED_COMMAND`, we introduce some sort of `TODOTXT_ENABLE_SED_SHIM`, which is disabled by default?

Then, we can do:

```suggestion
[ "$TODOTXT_SED_ENABLE_SHIM" = 1 ] && sed() {
```

That way, we can have better performance by default. If we go this route, then the variable `TODOTXT_SED_EXPORT_FOR_ADDONS` could be renamed to a more consistent `TODOTXT_SED_EXPORT_SHIM_FOR_ADDONS`. (and we would also have to check for this new variable before the `export -f sed` below)

Reply:
Maybe `SHIM` isn't the best word, because that could imply that it's some special `bin/shim`, which would even be worse for performance. But even with a better name, my idea is the same

Reply:
The ability to inject a custom `sed` command covers scenarios where people have a capable `sed` installed, but not accessible through `PATH`, or with a non-standard name. (Additionally, it enables nifty dev use cases, like injecting a `verboseSed` command that logs the sed arguments for troubleshooting.) To me, that's more flexibility for zero cost (versus an alternative yes/no flag).

If we go back to disabling the workaround by default, BSD and Busybox users are faced with a strange error message, and somehow have to find the workaround, like before. It's just a bit more convenient because the function is now packaged inside `todo.sh`, not a snippet they have to copy-and-paste into their configuration. But I'm afraid many won't easily find the flag in the config (who reads documentation and comments?), maybe google it, and then either find the GitHub issues (or just the older discussion mentioning the snippet).

I found a (somewhat clever :blush:) way to auto-detect the in-place capability in the `Makefile` and manipulate the default config (which is already done for the `TODO_DIR`, anyway). It's still quite a bit of code for a very minor optimization, but now that CI/CD performs `make [un]install` for every change, it feels acceptable to me. I have no idea how many users use which installation method, and how many are actually using BSD / Busybox with `todo.sh`; if someone else argues with simplicity and to scrap the whole workaround in this PR, I'd be fine with that, too.

Reply:
The ability to inject a custom sed command covers scenarios where people have a capable sed installed, but not accessible through PATH, or with a non-standard name. (Additionally, it enables nifty dev use cases, like injecting a verboseSed command that logs the sed arguments for troubleshooting.) To me, that's more flexibility for zero cost (versus an alternative yes/no flag).

:+1: 

> If we go back to disabling the workaround by default, BSD and Busybox users are faced with a strange error message, and somehow have to find the workaround, like before.

I really wouldn't want to force somebody to look up some random GitHub issue to get things to work if say they're on Busybox and get some sort of error. So yeah, better to enable the workaround by default

> I have no idea how many users use which installation method, and how many are actually using BSD / Busybox with todo.sh; if someone else argues with simplicity and to scrap the whole workaround in this PR, I'd be fine with that, too.

With the code already written, I see no reason to remove it! I think the auto-detection optimization is a good improvement